app/uk/gov/hmrc/directdebitupdateemailfrontend/config/AppConfig.scala

```diff
@@ -28,7 +30,8 @@ class AppConfig @Inject() (config: Configuration) {
 
   val authTimeoutSeconds: Int = config.get[FiniteDuration]("timeout-dialog.timeout").toSeconds.toInt
   val authTimeoutCountdownSeconds: Int = config.get[FiniteDuration]("timeout-dialog.countdown").toSeconds.toInt
-  val signOutUrl: String = BaseUrl.platformHost.getOrElse(config.get[String]("baseUrl.sign-out-local")) + "/bas-gateway/sign-out-without-state?continue=https%3A%2F%2Fwww.gov.uk"
+  def signOutUrl(implicit request: Request[_]): String = BaseUrl.platformHost.getOrElse(config.get[String]("baseUrl.sign-out-local")) +
+    s"/bas-gateway/sign-out-without-state?continue=${routes.SignOutController.signOut.absoluteURL()}"
```

Comment:
does the query parameter value get URL encoded automatically? Does it need to be or does it work when you tried this out?

Reply:
It does work yes when I test it out manually, locally.

It did occur to me though that the reason it works is that it's picking up the request from the previous call, ie. the GET which took the user to the page from where they clicked sign out. That's how it's constructing the base and the middle (prod route) of the url.

It works and it should always work since this method is only called from pages that must have been accessed via a request, but it seems a bit confused.

An alternative I'd prefer is to pull the base of the url from config somehow. But I couldn't work out how to gather the prod route without hard coding it. Where can I get the /direct-debit-verify-email bit of the url?

Reply:
if it works that ok - the bit I was worried about was the `routes.SignoutController.signout.absoluteUrl()` not being url encoded, but if it works then it's all good